firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java

```diff
@@ -0,0 +1,108 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.installations.local;
+
+import static android.content.ContentValues.TAG;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.util.Base64;
+import android.util.Log;
+import androidx.annotation.GuardedBy;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.google.firebase.FirebaseApp;
+import java.security.KeyFactory;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.X509EncodedKeySpec;
+
+public class IidStore {
+  private static final String IID_SHARED_PREFS_NAME = "com.google.android.gms.appid";
+  private static final String STORE_KEY_PUB = "|S||P|";
+  private static final String STORE_KEY_ID = "|S|id";
```

Comment:
both of these are missing the subtype part of the key format so they'll lose any subtype'd IDs

Reply:
Can you please provide some context on subtype. 

I saw that ```SUBTYPE_DEFAULT = "";``` in 
FirebaseInstanceId.java and this value is used for getting id from store.

Reply:
I don't honestly know much about this topic. Rayo will have more info on subtypes and whether they're important, I just want to know that we're explicitly ok with dropping non-default subtypes

Reply:
@andirayo Can you please take a look at this.

Thanks for pointing it out @ciarand

Reply:
When I looked at the IID Android code: this was the subtype default ``` SUBTYPE_DEFAULT = "";``` Hence, I skipped adding subtype.